documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1087 @@
+# and return them sorted by the publication year and title
+FROM com.acme.Book WHERE title like '%Data Grid%' ORDER BY publicationYear DESC, title ASC
+----
+
+==== Grouping and Aggregation
+
+{brandname} has the ability to group query results according to a set of grouping fields and construct aggregations of
+the results from each group by applying an aggregation function to the set of values that fall into each group.
+Grouping and aggregation can only be applied to projection queries (queries with one or more field in the SELECT clause).
+
+The supported aggregations are: avg, sum, count, max, min.
+
+The set of grouping fields is specified with the `GROUP BY` clause and the order used for defining grouping fields is
+not relevant. All fields selected in the projection must either be grouping fields
+or else they must be aggregated using one of the grouping functions described below. A projection field can be
+aggregated and used for grouping at the same time. A query that selects only grouping fields but no aggregation fields
+is legal.
+⁠
+Example: Grouping Books by author and counting them.
+[source,sql]
+----
+SELECT author, COUNT(title) FROM com.acme.Book WHERE title LIKE '%engine%' GROUP BY author
+----
+
+NOTE: A projection query in which all selected fields have an aggregation function applied and no fields are used for
+grouping is allowed. In this case the aggregations will be computed globally as if there was a single global group.
+
+==== Aggregations
+
+The following aggregation functions may be applied to a field: avg, sum, count, max, min
+
+
+* avg() - Computes the average of a set of numbers. Accepted values are primitive numbers and instances of _java.lang.Number_. The result is represented as _java.lang.Double_. If there are no non-null values the result is _null_ instead.
+* count() - Counts the number of non-null rows and returns a _java.lang.Long_. If there are no non-null values the result is _0_ instead.
+* max() - Returns the greatest value found. Accepted values must be instances of _java.lang.Comparable_. If there are no non-null values the result is _null_ instead.
+* min() - Returns the smallest value found. Accepted values must be instances of _java.lang.Comparable_. If there are no non-null values the result is _null_ instead.
+* sum() - Computes the sum of a set of Numbers. If there are no non-null values the result is _null_ instead. The following table indicates the return type based on the specified field.
+
+.Table sum return type
+|===
+|Field Type |Return Type
+
+|Integral (other than BigInteger)
+|Long
+
+|Float or Double
+|Double
+
+|BigInteger
+|BigInteger
+
+|BigDecimal
+|BigDecimal
+|===
+
+==== Evaluation of queries with grouping and aggregation
+
+Aggregation queries can include filtering conditions, like usual queries. Filtering can be performed in two stages: before
+and after the grouping operation. All filter conditions defined before invoking the _groupBy_ method will be applied
```

Comment:
I've hesitated a little to mention it because I don't want to go overboard with lots of little comments. I mentioned it in an earlier comment though so I'll bring it up. Code should use monospace formatting rather than italics so we're consistent in the docs. https://infinispan.org/docs/11.0.x/titles/contributing/contributing.html#documentation_guidelines_format_reference 

Also include "()" with method names. So _groupBy_ should change to `groupBy()`

Reply:
Ok, I will remove all italics